graphql_compiler/compiler/cypher_helpers.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+"""Helper functions for Cypher, such as generating unique vertex names within fold scopes."""
+from .helpers import FoldScopeLocation, Location
+
+
+def get_fold_scope_location_full_path_name(fold_scope_location):
+    """Return a unique name with the full traversal path to this FoldScopeLocation."""
+    # HACK(Leon): Get a unique name for each vertex in a fold traversal in Cypher.
+    # For FoldScopeLocation objects, get_location_name() only uses the first edge in the traversal,
+    # which doesn't work for Cypher because we need to explicitly label every intermediate vertex
+    # along that path.
+    # For other query languages like MATCH and Gremlin, the fold directive doesn't require
+    # us to name all the intermediate vertices on the path, which is why this is Cypher-specific.
+    if not (isinstance(fold_scope_location, FoldScopeLocation)):
```

Comment:
You technically don't need these parens either, but I'll stop nitpicking and merge this.